File: src/T5/lang/codigoGerado/teste12.js

function even(n) {
    if((n == 0)) {
        return [true];
    } else {
        return [odd((n - 1))[0]];
    }

}

function odd(n) {
    if((n == 0)) {
        return [false];
    } else {
        return [even((n - 1))[0]];
    }

}

function main() {
    r = even(3654)[0];
    if(r) {

        process.stdout.write('P'.toString());
        process.stdout.write('A'.toString());
        process.stdout.write('R'.toString());
    } else {

        process.stdout.write('I'.toString());
        process.stdout.write('M'.toString());
        process.stdout.write('P'.toString());
        process.stdout.write('A'.toString());
        process.stdout.write('R'.toString());
    }

}

main();  

File: src/T5/lang/codigoGerado/teste12.java

import java.util.ArrayList;
import java.util.Scanner;
public class teste12 {
    private static ArrayList<Object> _retornos = new ArrayList<Object>();

    static void even(int  n) {
         ArrayList<Object> _aux = new ArrayList<Object>();
        if((n == 0)) {
            _aux.add(true);
            for(Object o : _aux)
            {
                _retornos.add(o);
            }
            return;
        } else {
            _aux.add(((boolean)odd(String.valueOf(0), (n - 1))));
            for(Object o : _aux)
            {
                _retornos.add(o);
            }
            return;
        }

    }

     
    static Object even(String _i, int  n) {
        even(n); 
        Object _aux = _retornos.get(Integer.parseInt(_i));
        _retornos.clear();
        return _aux;
    }


    static void odd(int  n) {
         ArrayList<Object> _aux = new ArrayList<Object>();
        if((n == 0)) {
            _aux.add(false);
            for(Object o : _aux)
            {
                _retornos.add(o);
            }
            return;
        } else {
            _aux.add(((boolean)even(String.valueOf(0), (n - 1))));
            for(Object o : _aux)
            {
                _retornos.add(o);
            }
            return;
        }

    }

     
    static Object odd(String _i, int  n) {
        odd(n); 
        Object _aux = _retornos.get(Integer.parseInt(_i));
        _retornos.clear();
        return _aux;
    }


    static void main() {
        boolean r = false;
        r = ((boolean)even(String.valueOf(0), 3));
        if(r) {

            System.out.print('P');
            System.out.print('A');
            System.out.print('R');
        } else {

            System.out.print('I');
            System.out.print('M');
            System.out.print('P');
            System.out.print('A');
            System.out.print('R');
        }

    }



    public static void main(String args[]) {
        main();
    }
}